wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSessionBuilder.swift

```diff
@@ -0,0 +1,363 @@
+        guard
+            let appVersion,
+            let appLock,
+            let application,
+            let applicationStatusDirectory,
+            let contextStorage,
+            let coreCryptoProvider,
+            let coreDataStack,
+            let cryptoboxMigrationManager,
+            let e2eiActivationDateRepository,
+            let earService,
+            let flowManager,
+            let lastE2EIUpdateDateRepository,
+            let lastEventIDRepository,
+            let mediaManager,
+            let mlsConversationVerificationStatusUpdater,
+            let mlsService,
+            let observeMLSGroupVerificationStatusUseCase,
+            let proteusToMLSMigrationCoordinator,
+            let sharedUserDefaults,
+            let transportSession,
+            let updateMLSGroupVerificationStatusUseCase,
+            let useCaseFactory,
+            let userId
+        else {
+            fatalError("cannot build 'ZMUserSession' without required dependencies")
+        }
+
+        prepareCoreDataStack(
+            coreDataStack,
+            analytics: analytics
+        )
+
+        let userSession = ZMUserSession(
+            userId: userId,
+            transportSession: transportSession,
+            mediaManager: mediaManager,
+            flowManager: flowManager,
+            analytics: analytics,
+            application: application,
+            appVersion: appVersion,
+            coreDataStack: coreDataStack,
+            earService: earService,
+            mlsService: mlsService,
+            cryptoboxMigrationManager: cryptoboxMigrationManager,
+            proteusToMLSMigrationCoordinator: proteusToMLSMigrationCoordinator,
+            sharedUserDefaults: sharedUserDefaults,
+            useCaseFactory: useCaseFactory,
+            observeMLSGroupVerificationStatusUseCase: observeMLSGroupVerificationStatusUseCase,
+            appLock: appLock,
+            coreCryptoProvider: coreCryptoProvider,
+            lastEventIDRepository: lastEventIDRepository,
+            lastE2EIUpdateDateRepository: lastE2EIUpdateDateRepository,
+            e2eiActivationDateRepository: e2eiActivationDateRepository,
+            applicationStatusDirectory: applicationStatusDirectory,
+            updateMLSGroupVerificationStatusUseCase: updateMLSGroupVerificationStatusUseCase,
+            mlsConversationVerificationStatusUpdater: mlsConversationVerificationStatusUpdater,
+            contextStorage: contextStorage
+        )
+
+        return userSession
+    }
+
+    private func prepareCoreDataStack(
+        _ coreDataStack: CoreDataStack,
+        analytics: (any AnalyticsType)?
+    ) {
+        coreDataStack.syncContext.performGroupedBlockAndWait {
+            coreDataStack.syncContext.analytics = analytics
+            coreDataStack.syncContext.zm_userInterface = coreDataStack.viewContext
+        }
+        coreDataStack.viewContext.zm_sync = coreDataStack.syncContext
+    }
```

Comment:
I would expect that this preparation is done before the core data stack is passed to `ZMUserSession`, but not sure where to place it?